Add support for OpenCV 4

With Ubuntu 20.04 the stable version of opencv became opencv 4.
Under under Ubuntu 20.04 `cer` does not compile.

```
In file included from /home/ngenesio/robotology/cer/modules/faceExpression/main.cpp:26: 
/home/ngenesio/robotology/cer/modules/faceExpression/faceExpressionImage.hpp:5:10: fatal error: opencv/cv.h: No such file or directory 
   5 | #include <opencv/cv.h> 
     |          ^~~~~~~~~~~~~ 
compilation terminated.
In file included from /home/ngenesio/robotology/cer/modules/faceExpression/faceExpressionImage.cpp:3: 
/home/ngenesio/robotology/cer/modules/faceExpression/faceExpressionImage.hpp:5:10: fatal error: opencv/cv.h: No such file or directory 
   5 | #include <opencv/cv.h> 
     |          ^~~~~~~~~~~~~ 
compilation terminated.
make[2]: *** [modules/faceExpression/CMakeFiles/faceExpressionImage.dir/build.make:96: modules/faceExpression/CMakeFiles/faceExpressionImage.dir/main.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[2]: *** [modules/faceExpression/CMakeFiles/faceExpressionImage.dir/build.make:83: modules/faceExpression/CMakeFiles/faceExpressionImage.dir/faceExpressionImage.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:1802: modules/faceExpression/CMakeFiles/faceExpressionImage.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
/home/ngenesio/robotology/cer/cermod/faceDisplayServer/faceDisplayServer.cpp:14:10: fatal error: opencv/cv.h: No such file or directory 
  14 | #include <opencv/cv.h> 
     |          ^~~~~~~~~~~~~ 
compilation terminated.
make[2]: *** [cermod/faceDisplayServer/CMakeFiles/faceDisplayServer.dir/build.make:83: cermod/faceDisplayServer/CMakeFiles/faceDisplayServer.dir/faceDisplayServer.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
```

--- a/cermod/faceDisplayServer/faceDisplayServer.cpp
+++ b/cermod/faceDisplayServer/faceDisplayServer.cpp
@@ -11,13 +11,8 @@
 #include <sys/ioctl.h>
 #include <fcntl.h>
 #include <iostream>
-#include <opencv/cv.h>
-#include <opencv/cxcore.h>
-#include <opencv/highgui.h>
 
 #include <opencv2/core/mat.hpp>
-#include <opencv2/core/core.hpp>
-#include <opencv2/highgui/highgui.hpp>
 #include <opencv2/imgproc/imgproc.hpp>
 #include <opencv2/imgcodecs.hpp>
 
@@ -194,10 +189,11 @@ bool FaceDisplayServer::open(yarp::os::Searchable &config)
         return false;
     }
 
-    cv::cvtColor(imgHappyEye, imgHappyEye,CV_BGR2RGB);
-    cv::cvtColor(imgHappyBars,imgHappyBars,CV_BGR2RGB);
-    cv::cvtColor(imgSadEye,   imgSadEye,CV_BGR2RGB);
-    cv::cvtColor(imgSadBars,  imgSadBars,CV_BGR2RGB);
+
+    cv::cvtColor(imgHappyEye, imgHappyEye,cv::COLOR_BGR2RGB);
+    cv::cvtColor(imgHappyBars,imgHappyBars,cv::COLOR_BGR2RGB);
+    cv::cvtColor(imgSadEye,   imgSadEye,cv::COLOR_BGR2RGB);
+    cv::cvtColor(imgSadBars,  imgSadBars,cv::COLOR_BGR2RGB);
 
 //     yDebug() << "imgFace      cols " << face.cols << " rows " << face.rows << " channels " << face.channels();
 //     yDebug() << "imgHappyEye  cols " << imgHappyEye.cols << " rows " << imgHappyEye.rows << " channels " << imgHappyEye.channels() << " tot " << imgHappyEye.total() << " elemSize " << imgHappyEye.elemSize();
@@ -301,9 +297,9 @@ void FaceDisplayServer::run()
             {
                 yError() << "img not valid at index " << i;
                 return;
-            }    
+            }
+            cv::cvtColor(img[i], img[i], cv::COLOR_BGR2RGB);
 
-            cv::cvtColor(img[i], img[i], CV_BGR2RGB);
         }
 
         imageIdx = 0;
@@ -396,10 +392,8 @@ void FaceDisplayServer::run()
 
         cv::Mat image_tmp   = cv::imread(std::string(rootPath + "/homer.bmp").c_str(), cv::IMREAD_COLOR);
 
-
-        cv::cvtColor(image_1,image_1,CV_BGR2RGB);
-        cv::cvtColor(image_2,image_2,CV_BGR2RGB);
-
+        cv::cvtColor(image_1,image_1,cv::COLOR_BGR2RGB);
+        cv::cvtColor(image_2,image_2,cv::COLOR_BGR2RGB);
         cv::Mat *image_init, *image_final;
 
         image_init  = &image_1;
@@ -462,8 +456,7 @@ void FaceDisplayServer::run()
         }
 
         // Conver into BGR
-        cv::cvtColor(img,img,CV_BGR2RGB);
-
+        cv::cvtColor(img,img,cv::COLOR_BGR2RGB);
         int imageSize = IMAGE_WIDTH*IMAGE_HEIGHT*3;
         int rowSize = IMAGE_WIDTH*3;
         //  there is a dummy line between each image for easier understanding
@@ -494,8 +487,8 @@ void FaceDisplayServer::run()
             return;
         }
 
-        // Conver into BGR
-        cv::cvtColor(img,img,CV_BGR2RGB);
+        // Conver into BGRc
+        cv::cvtColor(img,img,cv::COLOR_BGR2RGB);
 
         int imageSize = IMAGE_WIDTH*IMAGE_HEIGHT*3;
         int rowSize = IMAGE_WIDTH*3;
@@ -830,7 +823,7 @@ void FaceDisplayServer::run()
                     }
 
 //                     Conver into BGR
-                    cv::cvtColor(imageFromFile,imageFromFile,CV_BGR2RGB);
+                    cv::cvtColor(imageFromFile,imageFromFile,cv::COLOR_BGR2RGB);
                     currentFace = &imageFromFile;
                 }
                 break;

--- a/cermod/faceDisplayServer/faceDisplayServer.h
+++ b/cermod/faceDisplayServer/faceDisplayServer.h
@@ -116,7 +116,7 @@ private:
 public:
     ImagePort(std::mutex &_mtx) : mtx(_mtx) { _fd=0; };
 
-    bool init(int fd)   { _fd = fd; }
+    bool init(int fd)   { _fd = fd;  return true;}
 
     virtual void onRead(yarp::sig::FlexImage& img)
     {

--- a/modules/faceExpression/faceExpressionImage.hpp
+++ b/modules/faceExpression/faceExpressionImage.hpp
@@ -2,7 +2,7 @@
 #include <string>
 #include <iostream>
 
-#include <opencv/cv.h>
+#include <opencv2/core/version.hpp>
 #include <opencv2/core/mat.hpp>
 
 #include <yarp/sig/Image.h>

